src/src/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -183,24 +183,6 @@ public void run() {
                                 view.loadUrl(loadUrl, headers);
                             }
                         });
-                    } catch (IllegalArgumentException e) {
-                        Logger.e(TAG, "Argument exception", e.getMessage(),
-                                ADALError.ARGUMENT_EXCEPTION, e);
-                        // It should return error code and finish the
-                        // activity, so that onActivityResult implementation
-                        // returns errors to callback.
-                        Intent resultIntent = new Intent();
-                        resultIntent.putExtra(
-                                        AuthenticationConstants.Browser.RESPONSE_AUTHENTICATION_EXCEPTION,
-                                        e);
-                        if (mRequest != null) {
-                            resultIntent.putExtra(
-                                    AuthenticationConstants.Browser.RESPONSE_REQUEST_INFO,
-                                    mRequest);
-                        }
-                        sendResponse(
```

Comment:
This illegalArgumentException is thrown when we process the challenge returned from server. I understand we should generally avoid catching illegalArgumentException especially in the case when checking for passed in arguments. But here is the case we hit conditional access flow, server send back the challenge, and adal need to process the challenge. We need to make sure the coming 401 challenge contains "headervalue". This has nothing to do with the client request, could potentially be the server error, do you still think we should crash and let developer fix his request? If we send back the error, then we know what's going wrong and will have the correlation id, and may know what's going wrong with the broker challenge. I agree the thrown exception could be changed though, it doesn't make a lot of sense to throw illegalArgument exception here.